mula/scheduler/server/server.py

```diff
@@ -567,6 +593,85 @@ def push_queue(self, queue_id: str, item: models.PrioritizedItem) -> Any:
 
         return models.PrioritizedItem(**p_item.model_dump())
 
+    def list_events(
+        self,
+        request: fastapi.Request,
+        task_id: Optional[str] = None,
+        type: Optional[str] = None,  # noqa: A002
+        context: Optional[str] = None,  # noqa: A002
+        event: Optional[str] = None,
+        min_timestamp: Optional[datetime.datetime] = None,
+        max_timestamp: Optional[datetime.datetime] = None,
+        offset: int = 0,
+        limit: int = 10,
+        filters: Optional[storage.filters.FilterRequest] = None,
+        item: Optional[models.Event] = None,
+    ) -> Any:
+        if item is not None and request.method == "POST":
+            created_event = fastapi.encoders.jsonable_encoder(self.create_event(item=item))
+
+            return fastapi.responses.JSONResponse(
+                status_code=status.HTTP_201_CREATED,
+                content=created_event,
+            )
+
+        if (min_timestamp is not None and max_timestamp is not None) and min_timestamp > max_timestamp:
+            raise fastapi.HTTPException(
+                status_code=fastapi.status.HTTP_400_BAD_REQUEST,
+                detail="min_timestamp cannot be greater than max_timestamp",
+            )
+
+        try:
+            results, count = self.ctx.datastores.event_store.get_events(
+                task_id=task_id,
+                type=type,
+                context=context,
+                event=event,
+                min_timestamp=min_timestamp,
+                max_timestamp=max_timestamp,
+                offset=offset,
+                limit=limit,
+                filters=filters,
+            )
+        except storage.filters.errors.FilterError as exc:
+            raise fastapi.HTTPException(
+                status_code=fastapi.status.HTTP_400_BAD_REQUEST,
+                detail=str(exc),
+            ) from exc
+        except ValueError as exc:
+            raise fastapi.HTTPException(
+                status_code=fastapi.status.HTTP_400_BAD_REQUEST,
+                detail=str(exc),
+            ) from exc
```

Comment:
These two clauses can be combined (`except FilterError, ValueError as exc: ...`)